keps/sig-auth/2579-psp-replacement/README.md

```diff
@@ -0,0 +1,1086 @@
+        - `spec.initContainers[*].securityContext.seLinuxOptions.type`
+    - **Allowed Values:**
+        - undefined/empty
+        - `container_t`
+        - `container_init_t`
+        - `container_kvm_t`
+
+- SELinuxOptions.User and SELinuxOptions.Role
+    - **Restricted Fields:**
+        - `spec.securityContext.seLinuxOptions.user`
+        - `spec.containers[*].securityContext.seLinuxOptions.user`
+        - `spec.initContainers[*].securityContext.seLinuxOptions.user`
+        - `spec.securityContext.seLinuxOptions.role`
+        - `spec.containers[*].securityContext.seLinuxOptions.role`
+        - `spec.initContainers[*].securityContext.seLinuxOptions.role`
+    - **Allowed Values:** undefined/empty
+
+- SELinuxOptions.Level
+    - Unrestricted.
+
+**Non-root Groups** - (restricted) This optional constraint will be omitted from the initial
+implementation.
+
+**Seccomp**
+- (restricted) Must be set to anything except `unconfined`. Same reasoning as AppArmor. If the default
+  value changes from `unconfined` (see https://github.com/kubernetes/enhancements/issues/2413), then
+  the requirement to set a profile will be lifted.
+- (baseline) May be unset; any profile except `unconfined` allowed.
+
+<<[UNRESOLVED]>>
+
+**BLOCKING**
+
+**Capabilities** - (baseline) Only the following capabilities may be added:
+- AUDIT_WRITE
+- CHOWN
+- DAC_OVERRIDE
+- FOWNER
+- FSETID
+- KILL
+- MKNOD
+- NET_BIND_SERVICE
+- SETFCAP
+- SETGID
+- SETPCAP
+- SETUID
+- SYS_CHROOT
+
+Notes:
+- This set is equal to the [docker default capability
+  set](https://docs.docker.com/engine/reference/run/#runtime-privilege-and-linux-capabilities) minus
+  `NET_RAW`.
+- The [OpenShift default capability
+  set](https://github.com/openshift/hypershift-toolkit/blob/148be6f31a365dbcdf1cbd647773f59ccbcc282a/assets/ignition/files/etc/crio/crio.conf#L87-L102)
+  also drops AUDIT_WRITE, MKNOD, SETFCAP, and SYS_CHROOT._
+- The allowed set for the restricted profile is implicitly empty (no capabilities), since
+  [Kubernetes does not support ambient
+  capabilities](https://github.com/kubernetes/kubernetes/issues/56374). If ambient capability
+  support is ever added, we may want to consider a more conservative allowed set for the restricted
+  profile.
+
+---
+
+Prior discussion:
+
+Kubernetes does not define the default set, but Docker does. This leaves us with 2 options:
+
+1. Hardcode the [docker default capability
+   set](https://docs.docker.com/engine/reference/run/#runtime-privilege-and-linux-capabilities), and
+   allow adding a capability in that set. This enables workloads to use an allowlist approach where
+   they drop all capabilities (`ALL`), and only explicitly add the ones they need. However, it means
+   that operators cannot effectively customize the default set (e.g. drop CAP_NET_RAW) through
+   runtime parameters.
+    - The drop all + add pattern is widely used. See
+      https://kubernetes.slack.com/archives/C0BP8PW9G/p1619042209240300 for discussion, and
```

Comment:
I can't tell what is being proposed for the `restricted` profile. does "The allowed set for the restricted profile is implicitly empty" mean that the `restricted` profile would not allow adding any capabilities, or is this proposing `baseline` and `restricted` behave the same?

Reply:
Under what is currently proposed, the restricted profile would allow the same capabilities as baseline - i.e. you can add capabilities from the allowed set. However, those capabilities are effectively dropped when the user changes to the non-root user, and `no_new_privs` blocks file capabilities, so none of this matters for restricted.

Do you think we should just say you can't add any capabilities for restricted, since doing so probably means you're misunderstanding what you're doing?